src/components/Swap/index.tsx

```diff
@@ -61,10 +66,17 @@ export default function Swap(props: SwapProps) {
 
   const focused = useHasFocus(wrapper)
 
+  // FIXME(kristiehuang): clarify when the integrator flow should open
```

Comment:
I think the integrator flow will need to be preventable by the integrator. You could do this with callbacks that propagate the initial click event:

```
<Wallet account={account} onClickConnectWallet={onClickConnectWallet} />

function Wallet({ account, onClickConnectWallet }) {
  const isConnected = Boolean(account)
  ...
  return isConnected ? (
      <Avatar account={account} />
    ) : (
      <ConnectWallet onClick={onClickConnectWallet} />
    )
}

function ConnectWallet({ onClick }) {
  const onConnect = useCallback((e: Event) => {
    onClick(e)
    if (e.defaultPrevented) return
    initiateOurOwnConnectWalletFlow()
  })
  ...
  return (<Button onClick={onConnect}>Connect Wallet</Button>)
}
```

Reply:
so the only way for an integrator to provide their own callback & also prevent our flow is by calling `e.preventDefault()` in their own callback?